### PR TITLE
lightning: check max compatible version (#760)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #760 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/787
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/787:release-4.0-a1e4b0154711
```

---

### What problem does this PR solve? 

Check TiDB cluster version before running TiDB-Lightning to avoid unexpected errors.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - Check TiDB cluster version before running TiDB-Lightning to avoid unexpected errors.

### CODE REVIEW

File: pkg/lightning/backend/importer.go [hunk 1]
```diff
@@ -39,9 +39,14 @@ const (
...
 var (
-	requiredTiDBVersion = *semver.New("2.1.0")
-	requiredPDVersion   = *semver.New("2.1.0")
-	requiredTiKVVersion = *semver.New("2.1.0")
+	// Importer backend is compatible with TiDB [2.1.0, 6.0.0).
+	requiredMinTiDBVersion = *semver.New("2.1.0")
+	requiredMinPDVersion   = *semver.New("2.1.0")
+	requiredMinTiKVVersion = *semver.New("2.1.0")
+	// TODO: bump max versions based on the version define in Makefile.
+	requiredMaxTiDBVersion = *semver.New("6.0.0")
```
Comment (glorv) [inline pkg/lightning/backend/importer.go:47]: for release-4.0 lightning, the max version should be `v5.0.0-alpha` 🤔？

File: pkg/lightning/backend/importer.go [hunk 2]
```diff
@@ -39,9 +39,14 @@ const (
...
-	requiredPDVersion   = *semver.New("2.1.0")
-	requiredTiKVVersion = *semver.New("2.1.0")
+	// Importer backend is compatible with TiDB [2.1.0, 5.0.0).
+	requiredMinTiDBVersion = *semver.New("2.1.0")
+	requiredMinPDVersion   = *semver.New("2.1.0")
+	requiredMinTiKVVersion = *semver.New("2.1.0")
+	// TODO: bump max versions based on the version define in Makefile.
+	requiredMaxTiDBVersion = *semver.New("5.0.0")
+	requiredMaxPDVersion   = *semver.New("5.0.0")
+	requiredMaxTiKVVersion = *semver.New("5.0.0")
```
Comment (glorv) [inline pkg/lightning/backend/importer.go:49]: `5.0.0-rc` or `5.0.0-beta` will pass this test since they are lower than `5.0.0` 🤔

Comment (overvenus) [reply pkg/lightning/backend/importer.go:49]: Indeed, I will address this issue in another PR.

Comment (overvenus) [reply pkg/lightning/backend/importer.go:49]: Addressed in https://github.com/pingcap/br/pull/806.

**Review by overvenus — APPROVED** [review]

**Pull request merged**